crates/anvil/src/evm.rs

```diff
@@ -31,6 +39,47 @@ pub fn inject_precompiles<DB, I>(
     }
 }
 
+/// A wrapper around the default ecrecover precompile that supports override via CheatsManager.
+pub fn custom_ecrecover(
+    input: &[u8],
+    gas_limit: u64,
+    cheats: Arc<CheatsManager>,
+) -> PrecompileResult {
+    const ECRECOVER_BASE: u64 = 3_000;
+    if gas_limit < ECRECOVER_BASE {
+        return Err(PrecompileError::OutOfGas);
+    }
+
+    let padded_input = right_pad::<128>(input);
+
+    // Only v = 27 or 28 is allowed
+    if !(padded_input[32..63].iter().all(|&b| b == 0) && matches!(padded_input[63], 27 | 28)) {
+        return Ok(PrecompileOutput::new(ECRECOVER_BASE, Bytes::new()));
+    }
+
+    let sig_bytes: [u8; 65] = {
+        let mut buf = [0u8; 65];
+        buf[..64].copy_from_slice(&padded_input[64..128]);
+        buf[64] = padded_input[63]; // recovery ID
+        buf
+    };
+
+    // Convert raw 65-byte sig into Signature
+    let sig = match Signature::try_from(&sig_bytes[..]) {
+        Ok(sig) => sig,
```

Comment:
what I'd like to do here is introduce an actual

`struct CheatEcrecover(Cheatsmanager)` struct then implement the precompile trait for this

we also want to check here if the cheatsmanager even has any signatures because if this is empty we dont need to do any of this overhead and can direcctly go to the fallback